### PR TITLE
feat: Add instance profile permissions to Karpenter IRSA policy

### LINKED ISSUE #433: bug; karpenter controller policy is missing required InstancePriofile persmissions indroduced in karpenter v0.32
## Description

Karpenter will auto-generate instance profiles as of [version v0.32](https://karpenter.sh/docs/upgrading/v1beta1-migration/#instanceprofile). The `karpenter_controller` policy in `modules/iam-role-for-service-accounts-eks/policies.tf` does not currently include statements for manipulating instance profiles. 

Updated karpenter policy can be found [in the karpenter repository](https://raw.githubusercontent.com/aws/karpenter/v0.32.1/website/content/en/preview/upgrading/v1beta1-controller-policy.json) 

- [x] ✋ I have searched the open/closed issues and my issue is not listed.

## Versions

- Module version [Required]: v5.30.1
- Terraform version: 1.6.3
- Provider version(s): hashicorp/aws 5.25.0
## Reproduction Code [Required]

Run the `karpenter_controller_irsa_role` module under `terraform-aws-iam/examples/iam-role-for-service-accounts-eks` ; 

```terraform
module "karpenter_controller_irsa_role" {
  source = "../../modules/iam-role-for-service-accounts-eks"

  role_name                          = "karpenter-controller"
  attach_karpenter_controller_policy = true

  karpenter_controller_cluster_name       = module.eks.cluster_name
  karpenter_controller_node_iam_role_arns = [module.eks.eks_managed_node_groups["default"].iam_role_arn]

  oidc_providers = {
    ex = {
      provider_arn               = module.eks.oidc_provider_arn
      namespace_service_accounts = ["karpenter:karpenter"]
    }
  }

  tags = local.tags
}
```
* Create a module from `iam-role-for-service-accounts-eks`  with `attach_karpenter_controller_policy = true` 
* Apply changes with terraform 
* Inspect resulting karpenter IAM policy. The policy is missing required instance profile actions

## Expected behavior

Expected a series of instance profile actions to be added to the karpenter role

## Actual behavior

Found no instance profile actions associated with the karpenter role/policy

## Additional context

Related Karpenter policy for the terraform module can be found [here](https://github.com/terraform-aws-modules/terraform-aws-iam/blob/2488f3085beeab2c3abd347771ccc8800d7699ac/modules/iam-role-for-service-accounts-eks/policies.tf#L610) 

Related Karpenter v0.32 policy can be found [here](https://raw.githubusercontent.com/aws/karpenter/v0.32.1/website/content/en/preview/upgrading/v1beta1-controller-policy.json)

### PR DESCRIPTION
**Pull request by abrabah**
fixes #433

### CODE REVIEW

File: modules/iam-role-for-service-accounts-eks/policies.tf [hunk 1]
```diff
@@ -691,6 +691,102 @@ data "aws_iam_policy_document" "karpenter_controller" {
...
+    }
+  }
+
+  statement {
+    actions = [
+      "iam:GetInstanceProfile"
+    ]
+    resources = ["*"]
+  }
+
```
Comment (ivankovnatsky) [inline modules/iam-role-for-service-accounts-eks/policies.tf:788]:
```suggestion
```

File: modules/iam-role-for-service-accounts-eks/policies.tf [hunk 2]
```diff
@@ -691,6 +691,102 @@ data "aws_iam_policy_document" "karpenter_controller" {
...
+      content {
+        test     = "StringEquals"
+        variable = condition.key
+        values   = [condition.value]
+      }
+    }
+  }
+
+  statement {
+
```
Comment (ivankovnatsky) [inline modules/iam-role-for-service-accounts-eks/policies.tf:719]:
```suggestion
```

File: modules/iam-role-for-service-accounts-eks/policies.tf [hunk 3]
```diff
@@ -691,6 +691,102 @@ data "aws_iam_policy_document" "karpenter_controller" {
...
+    actions = [
+      "iam:TagInstanceProfile"
+    ]
+
```
Comment (ivankovnatsky) [inline modules/iam-role-for-service-accounts-eks/policies.tf:723]:
```suggestion
```

File: modules/iam-role-for-service-accounts-eks/policies.tf [hunk 4]
```diff
@@ -691,6 +691,102 @@ data "aws_iam_policy_document" "karpenter_controller" {
...
+    }
+  }
+
+  statement {
+    actions = [
+      "iam:AddRoleToInstanceProfile",
+      "iam:RemoveRoleFromInstanceProfile",
+      "iam:DeleteInstanceProfile"
+    ]
+
```
Comment (ivankovnatsky) [inline modules/iam-role-for-service-accounts-eks/policies.tf:759]:
```suggestion
```

**Comment by ivankovnatsky** [comment]
Just tested your branch on my side, works as expected! Thanks.

**Review by bryantbiggs — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 5.32.0 :tada:

**Pull request merged**